mp/src/game/server/baseentity.cpp

```diff
@@ -5932,22 +5932,12 @@ void CBaseEntity::SetLocalOrigin( const Vector& origin )
 
 void CBaseEntity::SetLocalAngles( const QAngle& angles )
 {
-	// NOTE: The angle normalize is a little expensive, but we can save
-	// a bunch of time in interpolation if we don't have to invalidate everything
-	// and sometimes it's off by a normalization amount
-
-	// FIXME: The normalize caused problems in server code like momentary_rot_button that isn't
-	//        handling things like +/-180 degrees properly. This should be revisited.
-	//QAngle angleNormalize( AngleNormalize( angles.x ), AngleNormalize( angles.y ), AngleNormalize( angles.z ) );
-
 	// Safety check against NaN's or really huge numbers
+	// And actually normalize the angles rather than freeze the entity and spam warnings
 	if ( !IsEntityQAngleReasonable( angles ) )
 	{
-		if ( CheckEmitReasonablePhysicsSpew() )
-		{
-			Warning( "Bad SetLocalAngles(%f,%f,%f) on %s\n", angles.x, angles.y, angles.z, GetDebugName() );
-		}
-		AssertMsg( false, "Bad SetLocalAngles(%f,%f,%f) on %s\n", angles.x, angles.y, angles.z, GetDebugName() );
+		QAngle angleNormalize(AngleNormalize(angles.x), AngleNormalize(angles.y), AngleNormalize(angles.z));
+		SetLocalAngles(angleNormalize);
 		return;
```

Comment:
It may be better to just set `angles = angleNormalize;` here and instead of returning let the function continue with the new value.